### PR TITLE
Add configurable port for minikube mount

### LINKED ISSUE #11978: Add configurable port for minikube mount
The minikube mount command currently just uses a random free port to run the server on. If the user is using a firewall (definitely not me), this can block any requests from coming in, so mounting fails. Using a configurable port users can ensure the port is allowed through their firewall (again, not mine of course).

### PR DESCRIPTION
**Pull request by andriyDev**
fixes #11978.

Now users can specify what port to use for mounting (in addition to allowing random port assignment). Also, adds advice for timing out mounting to check firewall.

### CODE REVIEW

**Comment by andriyDev** [comment]
This PR is great to read each commit separately I'd say, hint hint

File: pkg/minikube/cluster/mount.go [hunk 1]
```diff
@@ -55,22 +57,25 @@ type mountRunner interface {
 }
 
 // Mount runs the mount command from the 9p client on the VM to the 9p server on the host
-func Mount(r mountRunner, source string, target string, c *MountConfig) error {
+func Mount(r mountRunner, source string, target string, c *MountConfig) {
 	if err := Unmount(r, target); err != nil {
-		return errors.Wrap(err, "umount")
+		exit.Error(reason.GuestMount, "mount failed", errors.Wrap(err, "umount"))
```
Comment (medyagh) [inline pkg/minikube/cluster/mount.go:62]: lets not exit in here and return an error type like this
https://github.com/medyagh/minikube/blob/0e665930e86e118976b5c6b3912aaed51ccfe20f/pkg/drivers/kic/oci/errors.go?plain=1#L37

**Review by medyagh — CHANGES_REQUESTED** [review]

File: pkg/minikube/cluster/mount.go (hunk 1 above)
Comment (andriyDev) [reply pkg/minikube/cluster/mount.go]: Done.

File: cmd/minikube/cmd/mount.go [hunk 2]
```diff
@@ -191,6 +193,7 @@ var mountCmd = &cobra.Command{
 
 		err = cluster.Mount(co.CP.Runner, ip.String(), vmPath, cfg)
 		if err != nil {
+			node.ExitIfFatal(err)
```
Comment (medyagh) [inline cmd/minikube/cmd/mount.go:196]: I dont think we should call the node package for this. node is meant for starting the node.

how about just checking the error type here and if that error type is do the right thing ?

**Review by medyagh — CHANGES_REQUESTED** [review]

File: pkg/minikube/cluster/mount.go [hunk 3]
```diff
@@ -54,19 +54,38 @@ type mountRunner interface {
...
 }
 
+const (
+	// MountErrorUnknown failed with unknown error
+	MountErrorUnknown = iota
+	// MountErrorConnect
+	MountErrorConnect
+)
+
+type MountError struct {
```
Comment (medyagh) [inline pkg/minikube/cluster/mount.go:64]: add comment for all exported types. (or dont make it exported if it hasn't been called outside this package

**Review by medyagh — CHANGES_REQUESTED** [review]

File: cmd/minikube/cmd/mount.go (hunk 2 above)
Comment (andriyDev) [reply cmd/minikube/cmd/mount.go]: Done.

File: pkg/minikube/cluster/mount.go (hunk 3 above)
Comment (andriyDev) [reply pkg/minikube/cluster/mount.go:65]: Done. It needs to be exported, else we can't check the ErrorType.

**Review by medyagh — APPROVED** [review]

**Pull request merged**